nomad/state/state_store_test.go

```diff
@@ -8087,6 +8087,34 @@ func TestStateStore_ClusterMetadataRestore(t *testing.T) {
 	require.Equal(now, out.CreateTime)
 }
 
+func TestStateStore_RestoreScalingPolicy(t *testing.T) {
+	t.Parallel()
+
+	state := testStateStore(t)
+	scalingPolicy := mock.ScalingPolicy()
+
+	restore, err := state.Restore()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	err = restore.ScalingPolicyRestore(scalingPolicy)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	restore.Commit()
+
+	ws := memdb.NewWatchSet()
+	out, err := state.ScalingPolicyByID(ws, scalingPolicy.ID)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	if !reflect.DeepEqual(out, scalingPolicy) {
+		t.Fatalf("Bad: %#v %#v", out, scalingPolicy)
+	}
```

Comment:
Use testify requires here as well?  Specially for diffing, testify provides a much better diff.